filter/src/lib.rs

```diff
@@ -49,6 +51,67 @@ enum Do {
     Httpbin(String),
 }
 
+impl fmt::Debug for Do {
```

Comment:
I thing semantically, you'd be better off with `impl Display` here.

Reply:
I tried `impl Display`, but ran into issues elsewhere because `Option<T>` where `T` is `Display` but not `Debug` is neither `Display` nor `Debug`, and I wasn't allowed to `impl Display for Option<Do>`.

Reply:
Ah, found the trick—`impl Display` for `Do`, and then a separate method to do `Option<Do> -> String`.